applications/API/serializers.py

```
@@ -0,0 +1,585 @@
+            category = typeresolve(raw_category)
+
+            data["category"] = InsultCategory.objects.get(category_key=category["category_key"])
+
+        return super().to_internal_value(data)
+
+    def to_representation(self, instance) -> Dict[str, Any]:  # pyrefly: ignore
+
+        representation = super().to_representation(instance)
+
+        # Use cached category lookup instead of additional DB query
+        validated_category = typeresolve(representation['category'])
```

Pylint (3.3.7)

[error] Undefined variable 'typeresolve'

(E0602)